### PR TITLE
z_demo_gt.c, 2/112 non matching

### PR DESCRIPTION
**Pull request by i82orbom**
I wanted to open this PR already to receive some feedback, there are 2 non matching which are almost matching, just some register issues.

This is my biggest decompiled actor so far, so please let me know if you have any suggestion for naming, etc...

### CODE REVIEW

**Review by fig02 — APPROVED** [review]
Implemented my review changes by hand

File: src/overlays/actors/ovl_Demo_Gt/z_demo_gt.h [hunk 1]
```diff
@@ -6,9 +6,22 @@
...
+    /* 0x0168 */ s32 drawConfig;
+    /* 0x016C */ Vec3s unk_16C;
+    /* 0x0172 */ s16 unk_172;
+    /* 0x0174 */ s16 unk_174;
+    /* 0x0178 */ Vec3i unk_178;
+    /* 0x0184 */ char unk_184[0x4];
+    /* 0x0188 */ Vec3i unk_188;
+    /* 0x0194 */ char unk_194[0x4];
+    /* 0x0198 */ Vec3i unk_198;
+    /* 0x01A4 */ char unk_1A4[0x4];
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Gt/z_demo_gt.h:24]: Considering their usage, these are probably just `s32[4]`
```suggestion
    /* 0x0178 */ s32 unk_178[4];
    /* 0x0188 */ s32 unk_188[4];
    /* 0x0198 */ s32 unk_198[4];
```

File: src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c [hunk 2]
```diff
@@ -9,7 +11,1794 @@ void DemoGt_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
+    func_80093D18(gfxCtx);
+    tempVec198 = &this->unk_198;
+    tempVec178 = &this->unk_178;
+    gSPSegment(POLY_OPA_DISP++, 0x08,
+               Gfx_TwoTexScrollEnvColor(gfxCtx, 0, 0U, tempVec198->x, 0x20, 0x40, 1, 0, tempVec198->y, 0x20, 0x40,
+                                        tempVec178->x, tempVec178->y, tempVec178->z, 0x80));
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(gfxCtx, "../z_demo_gt_part2.c", 485),
+              G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, D_06004F90);
+    gSPPopMatrixN(POLY_OPA_DISP++, G_MTX_MODELVIEW, 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c:894]:
```suggestion
    gSPPopMatrix(POLY_OPA_DISP++, G_MTX_MODELVIEW);
```

File: src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c [hunk 3]
```diff
@@ -9,7 +11,1794 @@ void DemoGt_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
+    Actor* actor;
+
+    if ((globalCtx->csCtx.frames > 1027) && (globalCtx->csCtx.frames < 1031)) {
+        pos.x = this->dyna.actor.posRot.pos.x;
+        pos.y = this->dyna.actor.posRot.pos.y + 247.0f;
+        pos.z = this->dyna.actor.posRot.pos.z;
+
+        if (cloudRing == NULL) {
+            cloudRing = DemoGt_SpawnCloudRing(globalCtx, &pos, 3);
+            return;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c:925]:
```suggestion
```

File: src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c [hunk 4]
```diff
@@ -9,7 +11,1794 @@ void DemoGt_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
+    Actor* actor;
+
+    if ((globalCtx->csCtx.frames > 997) && (globalCtx->csCtx.frames < 1001)) {
+        pos.x = this->dyna.actor.initPosRot.pos.x;
+        pos.y = this->dyna.actor.initPosRot.pos.y + 38.0f;
+        pos.z = this->dyna.actor.initPosRot.pos.z;
+
+        if (cloudRing == NULL) {
+            cloudRing = DemoGt_SpawnCloudRing(globalCtx, &pos, 4);
+            return;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c:948]:
```suggestion
```

File: src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c [hunk 5]
```diff
@@ -9,7 +11,1794 @@ void DemoGt_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
+void DemoGt_Draw3(DemoGt* this, GlobalContext* globalCtx) {
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+
+    OPEN_DISPS(gfxCtx, "../z_demo_gt_part3.c", 1026);
+
+    func_80093D18(gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(gfxCtx, "../z_demo_gt_part3.c", 1028),
+              G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, D_06007630);
+    gSPPopMatrixN(POLY_OPA_DISP++, G_MTX_MODELVIEW, 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c:1206]:
```suggestion
    gSPPopMatrix(POLY_OPA_DISP++, G_MTX_MODELVIEW);
```

File: src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c [hunk 6]
```diff
@@ -9,7 +11,1794 @@ void DemoGt_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
+        if (func_800C0D28(globalCtx2) == 0) {
+            func_80980F8C(this, globalCtx2);
+        }
+
+        Matrix_Pull();
+
+        func_80093D18(gfxCtx);
+        gSPMatrix(POLY_OPA_DISP++, sp60, (G_MTX_PUSH | G_MTX_LOAD) | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, D_06009610);
+        gSPPopMatrixN(POLY_OPA_DISP++, G_MTX_MODELVIEW, 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c:1328]:
```suggestion
        gSPPopMatrix(POLY_OPA_DISP++, G_MTX_MODELVIEW);
```

File: src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c [hunk 7]
```diff
@@ -9,7 +11,1794 @@ void DemoGt_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
+    if (func_800C0D28(globalCtx) == 0) {
+        func_80981458(this, globalCtx);
+    }
+
+    Matrix_Pull();
+
+    func_80093D18(gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, sp60, G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, D_06009610);
+    gSPPopMatrixN(POLY_OPA_DISP++, G_MTX_MODELVIEW, 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Gt/z_demo_gt.c:1447]:
```suggestion
    gSPPopMatrix(POLY_OPA_DISP++, G_MTX_MODELVIEW);
```

**Comment by fig02** [comment]
changes made

**Review by Roman971 — APPROVED** [review]

**Pull request merged**